proposed/extended-coding-style-guide.md

```diff
@@ -120,42 +120,34 @@ The PHP types and keywords `array`, `int`, `true`, `object`, `float`, `false`, `
 3. Declare Statements, Namespace, and Use Declarations
 --------------------------------------------
 
-When present, there MUST be one blank line after the `declare` statement(s)
-e.g. `declare(ticks=);`
-
-There MUST NOT be a blank line before declare statements such as those for strict
-types or ticks. They MUST be contained on the lines immediately following the
-opening tag (which must be on the first line when declare statement(s) are present).
-
-Each declare statement (e.g. `declare(ticks=);`) MUST be on its own line.
+The header of a PHP file may consist of a number of different blocks.  If present,
+each of the blocks below MUST be separated by a single blank line, and MUST NOT contain
+a blank line.  Each block MUST be in the order listed below, although blocks that are
+not relevant may be omitted.
+
+* Opening PHP tag <?php
+* File-level docblock.
```

Comment:
Probably but that's not what's enforced currently and I'm not sure it should not be enforced the other way around. Some reasearch:

### No blank lines

- [Yii](https://github.com/yiisoft/yii2/blob/master/framework/caching/ApcCache.php)
- [Aura](https://github.com/auraphp/Aura.Di/blob/3.x/src/Container.php)
- [Zend](https://github.com/zendframework/zend-expressive/blob/master/src/Application.php)
- [Slim](https://github.com/slimphp/Slim/blob/3.x/Slim/App.php)
- [Propel](https://github.com/propelorm/Propel2/blob/master/src/Propel/Common/Exception/SetColumnConverterException.php)
- [PPI](https://github.com/ppi/framework/blob/master/src/Console/Application.php)
- [Doctrine](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Event/ListenersInvoker.php)
- [CakePHP](https://github.com/cakephp/cakephp/blob/master/src/Collection/CollectionInterface.php)

### Blank lines

- [Laravel](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Support/ClassLoader.php)
- [Symfony](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/ConstraintViolation.php)
- [Composer](https://github.com/composer/composer/blob/master/src/Composer/Command/ArchiveCommand.php)